lib/floe/workflow/runner/kubernetes.rb

```diff
@@ -102,6 +102,64 @@ def cleanup(runner_context)
           delete_secret(secret) if secret
         end
 
+        def wait(timeout: nil, events: %i[create update delete])
+          retry_connection = true
+
+          begin
+            watcher = kubeclient.watch_pods(:namespace => namespace)
+
+            retry_connection = true
+
+            if timeout.to_i > 0
+              timeout_thread = Thread.new do
+                sleep(timeout)
+                watcher.finish
+              end
+            end
```

Comment:
not thrilled with this.
But it is a very common pattern (i.e.: feels like go code)

Reply:
Yeah it is my preference for any method call that could block indefinitely to take a timeout option, otherwise this is what you have to do.

`kubeclient.watch_pods(:timeout => 30.seconds)` would fix this :)